[2025-04-17T12:31:41.831+0530] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-04-17T12:31:41.845+0530] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: excel_data_analysis.process_excel_task manual__2025-04-17T06:56:42.066075+00:00 [queued]>
[2025-04-17T12:31:41.853+0530] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: excel_data_analysis.process_excel_task manual__2025-04-17T06:56:42.066075+00:00 [queued]>
[2025-04-17T12:31:41.853+0530] {taskinstance.py:2867} INFO - Starting attempt 2 of 3
[2025-04-17T12:31:41.907+0530] {taskinstance.py:2890} INFO - Executing <Task(PythonOperator): process_excel_task> on 2025-04-17 06:56:42.066075+00:00
[2025-04-17T12:31:41.912+0530] {standard_task_runner.py:72} INFO - Started process 52485 to run task
[2025-04-17T12:31:41.915+0530] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'excel_data_analysis', 'process_excel_task', 'manual__2025-04-17T06:56:42.066075+00:00', '--job-id', '9', '--raw', '--subdir', 'DAGS_FOLDER/file_manipulation.py', '--cfg-path', '/tmp/tmpy82albfn']
[2025-04-17T12:31:41.915+0530] {standard_task_runner.py:105} INFO - Job 9: Subtask process_excel_task
[2025-04-17T12:31:41.996+0530] {task_command.py:467} INFO - Running <TaskInstance: excel_data_analysis.process_excel_task manual__2025-04-17T06:56:42.066075+00:00 [running]> on host prakhar-8204
[2025-04-17T12:31:42.084+0530] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='excel_data_analysis' AIRFLOW_CTX_TASK_ID='process_excel_task' AIRFLOW_CTX_EXECUTION_DATE='2025-04-17T06:56:42.066075+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-17T06:56:42.066075+00:00'
[2025-04-17T12:31:42.085+0530] {logging_mixin.py:190} INFO - Task instance is in running state
[2025-04-17T12:31:42.086+0530] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2025-04-17T12:31:42.086+0530] {logging_mixin.py:190} INFO - Current task name:process_excel_task state:running start_date:2025-04-17 07:01:41.845810+00:00
[2025-04-17T12:31:42.086+0530] {logging_mixin.py:190} INFO - Dag name:excel_data_analysis and current dag run status:running
[2025-04-17T12:31:42.086+0530] {taskinstance.py:732} INFO - ::endgroup::
[2025-04-17T12:31:42.671+0530] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/dags/file_manipulation.py", line 159, in process_excel
    os.remove(top_gainers_chart_path)
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/top_gainers_chart.png'
[2025-04-17T12:31:42.676+0530] {logging_mixin.py:190} INFO - Task instance in failure state
[2025-04-17T12:31:42.676+0530] {logging_mixin.py:190} INFO - Task start:2025-04-17 07:01:41.845810+00:00 end:2025-04-17 07:01:42.676091+00:00 duration:0.830281
[2025-04-17T12:31:42.676+0530] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): process_excel_task> dag:<DAG: excel_data_analysis> dagrun:<DagRun excel_data_analysis @ 2025-04-17 06:56:42.066075+00:00: manual__2025-04-17T06:56:42.066075+00:00, state:running, queued_at: 2025-04-17 06:56:42.182309+00:00. externally triggered: True>
[2025-04-17T12:31:42.676+0530] {logging_mixin.py:190} INFO - Failure caused by [Errno 2] No such file or directory: '/tmp/top_gainers_chart.png'
[2025-04-17T12:31:42.677+0530] {taskinstance.py:1226} INFO - Marking task as UP_FOR_RETRY. dag_id=excel_data_analysis, task_id=process_excel_task, run_id=manual__2025-04-17T06:56:42.066075+00:00, execution_date=20250417T065642, start_date=20250417T070141, end_date=20250417T070142
[2025-04-17T12:31:42.996+0530] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-04-17T12:31:42.997+0530] {standard_task_runner.py:124} ERROR - Failed to execute job 9 for task process_excel_task ([Errno 2] No such file or directory: '/tmp/top_gainers_chart.png'; 52485)
Traceback (most recent call last):
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/airflowvenv/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/prakhar/Desktop/Bootcamp/Python/Airflow-Project/dags/file_manipulation.py", line 159, in process_excel
    os.remove(top_gainers_chart_path)
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/top_gainers_chart.png'
[2025-04-17T12:31:43.049+0530] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-04-17T12:31:43.139+0530] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-04-17T12:31:43.139+0530] {local_task_job_runner.py:245} INFO - ::endgroup::
